### ISSUE TITLE
Plugins

### ISSUE THREAD

**Issue by pfefferle** [jhass/nodeinfo#32]
Thinking about https://github.com/jhass/nodeinfo/issues/31

What do you think about adding a `plugins` or `extension` array, that has the same properties as `software` to list extensions that might enable some of the federation.

For example:

```json
"plugins": [{
    "name": "activitypub",
    "version": "0.5.0",
    "repository": "https://example.com",
},
{
    "name": "ostatus-for-wordpress",
    "version": "0.5.1",
    "repository": "https://example.com",
}],
```

like: https://fediverse.party/en/miscellaneous/

I am working on WordPress, where it might be usefull to know which plugins are used, and friendi.ca, hubzilla and gnu.social are also supporting plugins as far as I know.

**Comment by jhass** [comment]
I'm not sure tbh.

We already got `protocols` for knowing which nodes might communicate with each other and what kind of networks you're joining by signing up to the node. Beyond that I don't think this data is very comparable in a useful way across softwares. The next usecases I see would be informing node administrators about updates or polling usage stats for a particular plugin. Both of that probably actually is better done by a client specific to the base software. So my first intuition would be that doing it an the free-form metadata is enough.

**Comment by pfefferle** [comment]
Good point. I agree with your `protocols` argument, but we had some discussions, that the implementations might differ between different plugins. They might all support ActivityPub, OStatus or Diaspora* and the might all be compatible with Friendi.ca, Mastodon or Diaspora* but the might have different feature sets. Lets say, one only supports sharing posts, but no reactions or that one supports follwowing, the other only allows others to follow.

But you are right, using the `metadata` is perhaps a good first step to try!

**Comment by jhass** [comment]
Yeah well, we probably shouldn't have ActivityPub as a valid protocol but add like `activitypub/mastodon`, activitypub/nextcloud` etc. :P

**Comment by pfefferle** [comment]
good that it is an "Open Standard" 😁
